kore/src/Kore/Step/AxiomPatterns.hs

```diff
@@ -72,6 +77,17 @@ data RulePattern level variable = RulePattern
     }
     deriving (Eq, Show)
 
+instance Unparse (variable level) => Pretty (RulePattern level variable) where
+    pretty RulePattern { left, right, requires } =
```

Comment:
We use it everywhere, so I'm not necessarily asking for a change: with this type of extracting the fields it's easy to forget to update the Pretty instance when adding fields to the RulePattern. In many cases this would not be an issue, since we would actually want to use the new field in the same PR, but here we will only notice it only when we use the repl or something else that's not correlated with adding the field.

One suggestion would be to use `RulePattern _ _ _ _` for matching and to use the field getters inside.